src/analytics.js

```diff
@@ -435,13 +436,18 @@ window.addEventListener("beforeunload", async () => {
 	await Analytics.flushEvents();
 });
 
-// Expose userland API.
-window.Altis.Analytics.updateEndpoint = Analytics.updateEndpoint;
-window.Altis.Analytics.record = (type, data = {}) =>
-	Analytics.record(
-		type,
-		{
-			attributes: getAttributes(data.attributes || {}),
-			metrics: getMetrics(data.metrics || {})
-		}
-	);
+// Expose userland API. Return empty functions if the Noop field is true.
+if (Noop) {
+	Analytics.updateEndpoint = () => { };
+	Analytics.record = () => { };
+} else {
+	window.Altis.Analytics.updateEndpoint = Analytics.updateEndpoint;
+	window.Altis.Analytics.record = (type, data = {}) =>
+		Analytics.record(
+			type,
+			{
+				attributes: getAttributes(data.attributes || {}),
+				metrics: getMetrics(data.metrics || {})
+			}
+		);
+}
```

Comment:
Make sure to configure your editor to add a blank line at the end of your files, it's a more of a habit from writing shell scripts so a program know when to terminate reading a file but it's also part of our coding standards.